catkin_ws/src/kpr_arm_control/src/arm_control/arm_control.cpp

```diff
@@ -19,35 +38,61 @@ using namespace ros;
  * @return True if the movement succeeded, false otherwise.
  */
 bool moveArmTo(geometry_msgs::Pose targetPose) {
-	if (!moveit_consts::move_group_ptr) {
+	if (!move_group_ptr) {
 		ROS_ERROR("The Move Group does not exist!");
 		return false;
 	}
+	double distance = sqrt(pow(targetPose.position.x,2) +
+			pow(targetPose.position.y,2) + pow(targetPose.position.z,2));
 
-	targetPose.orientation.w = 1.0;
-	moveit_consts::move_group_ptr->setPoseTarget(targetPose);
+	move_group_ptr->setPoseTarget(targetPose);
 
 	moveit::planning_interface::MoveGroup::Plan my_plan;
-	bool success = moveit_consts::move_group_ptr->plan(my_plan);
+	bool success = move_group_ptr->plan(my_plan);
 
 	if (!success) {
 		ROS_ERROR("Could not plan a move; move is probably impossible.");
-		return false;
+		if (distance > 0.8) {
+			ROS_ERROR("Move failed because location was out of reach.");
+		} else if (distance < 0.25) {
+			ROS_ERROR("Move failed because location was probably too close to the base");
+		}
+		return success;
 	}
-	moveit_consts::move_group_ptr->move();
+	move_group_ptr->move();
 
 	return success;
 }
 
-bool startGrip(CucumberContainer cucumber) {
-	return false;
+bool startGrip() {
+	ur_msgs::SetIO io_message;
+	io_message.request.fun = io_message.request.FUN_SET_DIGITAL_OUT;
+	io_message.request.pin = GRIPPER_PIN;
+	io_message.request.state = GRIPPER_CLOSE;
+	return io_state_client.call(io_message) && io_message.response.success;
 }
 
-void cut() {
-	//TODO Code to cut the cucumber.
+bool cut() {
+	ur_msgs::SetIO io_message_close;
+	io_message_close.request.fun = io_message_close.request.FUN_SET_DIGITAL_OUT;
+	io_message_close.request.pin = CUTTER_PIN;
+	io_message_close.request.state = CUTTER_CLOSE;
+	if (!(io_state_client.call(io_message_close) && io_message_close.response.success)) {
+		return false;
+	}
+
+	//TODO Not sure if a timeout should be here.
```

Comment:
I think we should separate the closing and opening of the cutter, so that the we can read out its sensors and make a decision based on them.